stock_reserve/model/stock_reserve.py

```diff
@@ -200,3 +179,27 @@ def open_move(self):
             views=[(view_id, "form")], res_id=self.move_id.id,
         )
         return action_dict
+
+    def write(self, vals):
+        res = super().write(vals)
+        if (
+            "product_uom_qty" in vals
+            and self.state == "partially_available"
+            and self.product_id.virtual_available >= 0
+        ):
+            self.reserve()
+        return res
+
+    @api.model
+    def assign_waiting_confirmed_reserve_moves(self):
+        reservations_to_assign = self.search(
+            [
+                ("state", "in", ["confirmed", "waiting", "partially_available"]),
```

Comment:
Maybe externalize domain retrieval in a function ?

Reply:
> @rousseldenis changes done
> 
> > Maybe externalize domain retrieval in a function ?
> 
> I don't know if you mean what I have done

Nearly, in order to allow inheritance, domain retrieval is better in a separate function, but for the whole one.

Like:

self.search(self._get_reservations_to_assign_domain())